crates/core/src/bcudp/crc.rs

```diff
@@ -0,0 +1,79 @@
+    0x63, 0xE3, 0x84, 0x3B, 0x64, 0x94, 0x3E, 0x6A, 0x6D, 0xD, 0xA8, 0x5A, 0x6A, 0x7A, 0xB, 0xCF,
+    0xE, 0xE4, 0x9D, 0xFF, 0x9, 0x93, 0x27, 0xAE, 0x0, 0xA, 0xB1, 0x9E, 0x7, 0x7D, 0x44, 0x93, 0xF,
+    0xF0, 0xD2, 0xA3, 0x8, 0x87, 0x68, 0xF2, 0x1, 0x1E, 0xFE, 0xC2, 0x6, 0x69, 0x5D, 0x57, 0x62,
+    0xF7, 0xCB, 0x67, 0x65, 0x80, 0x71, 0x36, 0x6C, 0x19, 0xE7, 0x6, 0x6B, 0x6E, 0x76, 0x1B, 0xD4,
+    0xFE, 0xE0, 0x2B, 0xD3, 0x89, 0x5A, 0x7A, 0xDA, 0x10, 0xCC, 0x4A, 0xDD, 0x67, 0x6F, 0xDF, 0xB9,
+    0xF9, 0xF9, 0xEF, 0xBE, 0x8E, 0x43, 0xBE, 0xB7, 0x17, 0xD5, 0x8E, 0xB0, 0x60, 0xE8, 0xA3, 0xD6,
+    0xD6, 0x7E, 0x93, 0xD1, 0xA1, 0xC4, 0xC2, 0xD8, 0x38, 0x52, 0xF2, 0xDF, 0x4F, 0xF1, 0x67, 0xBB,
+    0xD1, 0x67, 0x57, 0xBC, 0xA6, 0xDD, 0x6, 0xB5, 0x3F, 0x4B, 0x36, 0xB2, 0x48, 0xDA, 0x2B, 0xD,
+    0xD8, 0x4C, 0x1B, 0xA, 0xAF, 0xF6, 0x4A, 0x3, 0x36, 0x60, 0x7A, 0x4, 0x41, 0xC3, 0xEF, 0x60,
+    0xDF, 0x55, 0xDF, 0x67, 0xA8, 0xEF, 0x8E, 0x6E, 0x31, 0x79, 0xBE, 0x69, 0x46, 0x8C, 0xB3, 0x61,
+    0xCB, 0x1A, 0x83, 0x66, 0xBC, 0xA0, 0xD2, 0x6F, 0x25, 0x36, 0xE2, 0x68, 0x52, 0x95, 0x77, 0xC,
+    0xCC, 0x3, 0x47, 0xB, 0xBB, 0xB9, 0x16, 0x2, 0x22, 0x2F, 0x26, 0x5, 0x55, 0xBE, 0x3B, 0xBA,
+    0xC5, 0x28, 0xB, 0xBD, 0xB2, 0x92, 0x5A, 0xB4, 0x2B, 0x4, 0x6A, 0xB3, 0x5C, 0xA7, 0xFF, 0xD7,
+    0xC2, 0x31, 0xCF, 0xD0, 0xB5, 0x8B, 0x9E, 0xD9, 0x2C, 0x1D, 0xAE, 0xDE, 0x5B, 0xB0, 0xC2, 0x64,
+    0x9B, 0x26, 0xF2, 0x63, 0xEC, 0x9C, 0xA3, 0x6A, 0x75, 0xA, 0x93, 0x6D, 0x2, 0xA9, 0x6, 0x9,
+    0x9C, 0x3F, 0x36, 0xE, 0xEB, 0x85, 0x67, 0x7, 0x72, 0x13, 0x57, 0x0, 0x5, 0x82, 0x4A, 0xBF,
+    0x95, 0x14, 0x7A, 0xB8, 0xE2, 0xAE, 0x2B, 0xB1, 0x7B, 0x38, 0x1B, 0xB6, 0xC, 0x9B, 0x8E, 0xD2,
+    0x92, 0xD, 0xBE, 0xD5, 0xE5, 0xB7, 0xEF, 0xDC, 0x7C, 0x21, 0xDF, 0xDB, 0xB, 0xD4, 0xD2, 0xD3,
+    0x86, 0x42, 0xE2, 0xD4, 0xF1, 0xF8, 0xB3, 0xDD, 0x68, 0x6E, 0x83, 0xDA, 0x1F, 0xCD, 0x16, 0xBE,
+    0x81, 0x5B, 0x26, 0xB9, 0xF6, 0xE1, 0x77, 0xB0, 0x6F, 0x77, 0x47, 0xB7, 0x18, 0xE6, 0x5A, 0x8,
+    0x88, 0x70, 0x6A, 0xF, 0xFF, 0xCA, 0x3B, 0x6, 0x66, 0x5C, 0xB, 0x1, 0x11, 0xFF, 0x9E, 0x65,
+    0x8F, 0x69, 0xAE, 0x62, 0xF8, 0xD3, 0xFF, 0x6B, 0x61, 0x45, 0xCF, 0x6C, 0x16, 0x78, 0xE2, 0xA,
+    0xA0, 0xEE, 0xD2, 0xD, 0xD7, 0x54, 0x83, 0x4, 0x4E, 0xC2, 0xB3, 0x3, 0x39, 0x61, 0x26, 0x67,
+    0xA7, 0xF7, 0x16, 0x60, 0xD0, 0x4D, 0x47, 0x69, 0x49, 0xDB, 0x77, 0x6E, 0x3E, 0x4A, 0x6A, 0xD1,
+    0xAE, 0xDC, 0x5A, 0xD6, 0xD9, 0x66, 0xB, 0xDF, 0x40, 0xF0, 0x3B, 0xD8, 0x37, 0x53, 0xAE, 0xBC,
+    0xA9, 0xC5, 0x9E, 0xBB, 0xDE, 0x7F, 0xCF, 0xB2, 0x47, 0xE9, 0xFF, 0xB5, 0x30, 0x1C, 0xF2, 0xBD,
+    0xBD, 0x8A, 0xC2, 0xBA, 0xCA, 0x30, 0x93, 0xB3, 0x53, 0xA6, 0xA3, 0xB4, 0x24, 0x5, 0x36, 0xD0,
+    0xBA, 0x93, 0x6, 0xD7, 0xCD, 0x29, 0x57, 0xDE, 0x54, 0xBF, 0x67, 0xD9, 0x23, 0x2E, 0x7A, 0x66,
+    0xB3, 0xB8, 0x4A, 0x61, 0xC4, 0x2, 0x1B, 0x68, 0x5D, 0x94, 0x2B, 0x6F, 0x2A, 0x37, 0xBE, 0xB,
+    0xB4, 0xA1, 0x8E, 0xC, 0xC3, 0x1B, 0xDF, 0x5, 0x5A, 0x8D, 0xEF, 0x02, 0x2D,
```

Comment:
Is there not a crate that has this table already? What variant of CRC is this?

Reply:
I have no idea what the CRC is, most of this code comes from the python program [here](https://github.com/jdillenkofer/camera_proxy) I had a look into this but it didn't seem to match the CRC uses for UDP protocol and CRCs are not my area of expertise so I am usure where it is from.

Reply:
Did some google and maybe [this](https://github.com/srijs/rust-crc32fast/blob/master/src/table.rs) one will work :)

Reply:
Tried every algorithm in the CRC crate.

The current method provides this:
- Should BE: 1115833684

While every u32 method in the crc crate provides a different value:
- CRC_32_ISCSI was 2699336615
- CRC_32_AIXM was 103931297
- CRC_32_AUTOSAR was 2232806661
- CRC_32_BASE91_D was 1196129179
- CRC_32_BZIP2 was 2046554463
- CRC_32_CD_ROM_EDC was 419968025
- CRC_32_CKSUM was 3152376032
- CRC_32_ISO_HDLC was 1090306792
- CRC_32_JAMCRC was 3204660503
- CRC_32_MPEG_2 was 2248412832
- CRC_32_XFER was 3116207526

Reply:
Ok I i figured it out. It's a crc with a polynomial of 0x04c11db7 like CRC_32_ISO_HDLC BUT it has an init value of 0x00000000 (instead of 0xffffffff) and an xorout of 0x00000000 (instead of 0xffffffff)

Two ways to implement this:

1. Use the crc crate with the poly and parameters manually supplied.
2. Use the fastcrc crate but adjust the init value and undo the final xorout

Reply:
I will use the fastcrc crate as it:
- uses a fixed table lookup (instead of calculating from the polynomial)
- Uses sse or aarch64 acceletation were possible and falls back to 4 bytes at a time were not